include/irr/asset/IDescriptorSetLayout.h

```diff
@@ -103,21 +105,22 @@ class IDescriptorSetLayout : public virtual core::IReferenceCounted
 				if (bnd.type==EDT_COMBINED_IMAGE_SAMPLER && bnd.samplers)
 					immSamplerCount += bnd.count;
 			}
-			m_samplers = core::make_refctd_dynamic_array<core::smart_refctd_dynamic_array<core::smart_refctd_ptr<SamplerType>>>(immSamplerCount);
+			m_samplers = core::make_refctd_dynamic_array<core::smart_refctd_dynamic_array<core::smart_refctd_ptr<sampler_type> > >(immSamplerCount);
 
 			size_t immSamplersOffset = 0u;
 			for (size_t i = 0ull; i < bndCount; ++i)
 			{
 				auto& bnd_out = m_bindings->operator[](i);
 				const auto& bnd_in = _begin[i];
 
+				bnd_out.binding = bnd_in.binding;
```

Comment:
this is why it was all crashing, every single descriptor was set to binding =0 , knock on effect was wrong sized static arrays